src/Microsoft.ML.Dnn/ImageClassificationTransform.cs

```diff
@@ -200,16 +220,16 @@ public ImageProcessor(ImageClassificationTransformer transformer)
                 _imagePreprocessingRunner.AddOutputs(transformer._resizedImageTensorName);
             }
 
-            public Tensor ProcessImage(string path)
+            public Tensor ProcessImage(VBuffer<byte> imgBuf)
```

Comment:
>imgBuf [](start = 53, length = 6)

Can this be passed either with `in` or with `ref`?

Reply:
@yaeld@microsoft.com What will be the advantage? It is not like we are manipulating the VBuffer here, instead we are just reading the contents from the vbuffer and passing to tensorflow to decode and preprocess the image. I understand structs are pass by value but here the memory is allocated on the stack.

---
In reply to: [327492184](https://github.com/dotnet/machinelearning/pull/4242#discussion_r327492184) [](ancestors = 327492184)